reference/forms/types/options/empty_data.rst.inc

```diff
@@ -10,22 +10,34 @@ empty_data
 DEFAULT_PLACEHOLDER
 
 This option determines what value the field will return when the submitted
-value is empty.
+value is empty. It does not set an initial value if none is provided when
+the form is rendered in a view (see ``data`` or ``placeholder`` options).
 
-But you can customize this to your needs. For example, if you want the
-``gender`` choice field to be explicitly set to ``null`` when no value is
-selected, you can do it like this::
+It helps you handling form submission and you can customize this to your needs.
+For example, if you want the ``gender`` choice field to be explicitly set to ``Male``
+when no value is selected, you can do it like this::
 
     $builder->add('gender', 'choice', array(
         'choices' => array(
```

Comment:
This example also needs to be changed (ref https://github.com/silexphp/Silex/pull/1300#event-504818302).

And it make no sense to use `ChoiceType` wich is the most complicated example.

We should override it for this type to make it even clearer as it is on its own the most confusing and principal cause of issues.